### ISSUE TITLE
Password Generator results in invalid RabbitMQ Connection Strings sometimes

### ISSUE THREAD

**Issue by mohdali** [dotnet/aspire#3100]
The Password Generator for RabbitMQ adds two special characters to the password. This results in invalid Uris in the connection string if special characters such as `+` are randomly selected by the generator.

I would suggest to either remove special characters from the generator or exclude characters that could result in invalid URLs.

Example of invalid URL:
`amqp://guest:ZUntK+Bgje40gD{R@localhost:59684`

**Comment by davidfowl** [comment]
@eerhardt Did we fix this in p5?

**Comment by eerhardt** [comment]
Yes, this was fixed in https://github.com/dotnet/aspire/pull/2717. We choose to not use special characters at all in RabbitMQ generated passwords.

https://github.com/dotnet/aspire/blob/941a37041a8dbfe2a7c6b01a50633866618c378d/src/Aspire.Hosting.RabbitMQ/RabbitMQServerResource.cs#L28-L29
